lib/OpenQA/Git/MaintenanceOutage.pm

```diff
@@ -0,0 +1,37 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package OpenQA::Git::MaintenanceOutage;
+use Mojo::Base -strict, -signatures;
+use Mojo::File 'path';
+
+# Remove or rename these "constants" in favor of a dynamic path
+use constant MAX_DURATION_SECONDS => 1800;
+
+use constant {
+    SKIP => 'SKIP',
+    FAIL => 'FAIL',
+};
+
+sub _state_file ($app) {
+    path($ENV{OPENQA_GIT_SERVER_MAINTENANCE_FILE} // ($app->config->{global}->{cache_dir} // '/var/lib/openqa/cache')
+          . ($ENV{OPENQA_GIT_SERVER_MAINTENANCE_FILE} ? '' : '/gitlab_maintenance.lock'));
+}
+
+sub remove_state_file ($app) {
+    my $file = _state_file($app);
+    return unless -f $file;
+    $file->remove;
+}
```

Comment:
`Mojo::File::remove` doesn't fail if the file doesn't exist so the additional condition/check isn't necessary:

```suggestion
sub remove_state_file ($app) { _stat_file($app)->remove }
```